bin/buckinghamshire/create-parish-bodies

```diff
@@ -0,0 +1,90 @@
+# The key is the mapit name, value is the desired display name.
+# Anything not in this list will default to "$mapit_name Parish Council"
+my $name_mappings = {
+    'Addington' => 'Addington (meeting)',
+    'Amersham' => 'Amersham Town Council',
+    'Aston Sandford' => 'Aston Sandford (meeting)',
+    'Aylesbury' => 'Aylesbury Town Council',
+    'Barton Hartshorn' => 'Barton Hartshorn (meeting)',
+    'Beaconsfield' => 'Beaconsfield Town Council',
+    'Biddlesden' => 'Biddlesden (meeting)',
+    'Bledlow-cum-Saunderton' => 'Bledlow-Cum-Saunderton Parish Council',
+    'Buckingham' => 'Buckingham Town Council',
+    'Chesham' => 'Chesham Town Council',
+    'Chetwode' => 'Chetwode (meeting)',
+    'Cholesbury-cum-St Leonards' => 'Cholesbury-Cum-St Leonards Parish Council',
+    'Coldharbour' => 'Coldharbour',
+    'Creslow' => 'Creslow (meeting)',
+    'Dinton-with-Ford and Upton' => 'Dinton-with-Ford and Upton',
+    'Dorton' => 'Dorton Parish Council (meeting)',
+    'Drayton Beauchamp' => 'Drayton Beauchamp (meeting)',
+    'Dunton' => 'Dunton (meeting)',
+    'Edlesborough' => 'Edlesborough Northall and Dagnall Parish Council',
+    'Gerrards Cross' => 'Gerrards Cross Town Council',
+    'Hedsor' => 'Hedsor (Meeting)',
+    'Hoggeston' => 'Hoggeston (meeting)',
+    'Kingsey' => 'Kingsey (meeting)',
+    'Kingswood' => 'Kingswood (meeting)',
+    'Longwick-cum-Ilmer' => 'Longwick cum Ilmer Parish Council',
+    'Marlow' => 'Marlow Town Council',
+    'Nether Winchendon' => 'Nether Winchendon (meeting)',
+    'Poundon' => 'Poundon (meeting)',
+    'Princes Risborough' => 'Princes Risborough Town Council',
+    'Shalstone' => 'Shalstone (meeting)',
+    'Thornton' => 'Thornton (meeting)',
+    'Water Stratford' => 'Water Stratford (meeting)',
+    'Wingrave with Rowsham' => 'Wingrave-with-Rowsham Parish Council',
+    'Winslow' => 'Winslow Town Council',
+    'Wooburn' => 'Wooburn and Bourne End Parish Council',
+    'Woodham' => 'Woodham (meeting)',
+    'Wotton Underwood' => 'Wotton Underwood (meeting)',
+};
+
+my $parent_id = "163793"; # Buckinghamshire Council Unitary Authority on Mapit
+
+# Fetch the list of parishes from MapIt
+my $mapit_areas = mySociety::MaPit::call('area', "$parent_id/children", type => 'CPC');
+my @areas = values %$mapit_areas;
+
+my $db = FixMyStreet::DB->schema->storage;
+$db->txn_do(sub {
+    foreach my $area (@areas) {
+        my $id = $area->{id};
+        my $name = $name_mappings->{$area->{name}} || "$area->{name} Parish Council";
+
+        my $body_rs = FixMyStreet::DB->resultset("Body")->search({
+            id => $id,
+            name => $name,
+        });
+
+        if ($body_rs->count > 0) {
+            print "Existing body found for $name, skipping...\n";
+            next;
+        }
+
+        print "Creating body $name ($id)...";
+
+        my $body = $body_rs->create({ send_method => 'Noop' });
+        $body->body_areas->find_or_create({ area_id => $id });
```

Comment:
I guess this won't be going on live before other code is ready, wondered if needed to consider a way to not add this body_area (or add it later on), so that it e.g. could be run on live site without the bodies becoming active as they didn't cover an area, iyswim.

Reply:
Yeah, I guess ideally I'd do this in add-parish-categories when adding the categories to the parish, since we probably want to make it active at that point, maybe? But I don't have easy access to the mapit ID in that script.

If I add the body_areas entry but the parish has no categories, is it still considered to be active?

Reply:
Even when you've added the categories, I guess it depends if the other code is all ready for reports to go to them correctly that we discussed (but yes, assuming that, that sounds okay).

If you add the body_areas but have no categories, then when you click the map on FMS and it finds one body with categories and one without, it will put the other body in missing_details_bodies and display the error message about we're missing details for that body and asking them to contact us. You could presumably put in an exception to ignore/not do this in some manner, though would be good to think of a better way....

...ooh if you add the bodies marked as deleted! That might work? Then you can undelete them when ready. Sorry for not thinking of that before (if it works!).